File: EletricGo/Domain/Trucks/Truck.cs
using EletricGo.Domain.Shared;
using EletricGo.Domain.Trucks.ValueObjects;
using EletricGo.Domain.Trucks.DTO;

namespace EletricGo.Domain.Trucks 
{
    public class Truck : Entity<TruckID>
    {

        private TruckID truckID { get; set;}
        private Tare tare { get; set; }
        private Capacity capacity { get; set; }
        private MaxBatteryCapacity maxBatteryCapacity { get; set; }
        private Autonomy autonomy { get; set; }
        private FastChargeTime fastChargeTime { get; set; }

        public Truck(TruckID truckID, Tare tare, Capacity capacity, MaxBatteryCapacity maxBatteryCapacity, Autonomy autonomy, FastChargeTime fastChargeTime)
        {
            this.Id = new TruckID(Guid.NewGuid());
            this.truckID = truckID;
            this.tare = tare;
            this.capacity = capacity;
            this.maxBatteryCapacity = maxBatteryCapacity;
            this.autonomy = autonomy;
            this.fastChargeTime = fastChargeTime;
        }

        public Truck(TruckDTO truckDTO)
        {
            this.truckID = new TruckID(truckDTO.truckID);
            this.tare = new Tare(truckDTO.tare);
            this.capacity = new Capacity(truckDTO.capacity);
            this.maxBatteryCapacity = new MaxBatteryCapacity(truckDTO.maxBatteryCapacity);
            this.autonomy = new Autonomy(truckDTO.autonomy);
            this.fastChargeTime = new FastChargeTime(truckDTO.fastChargeTime);
        }

        public TruckDTO toTruckDTO()
        {
            return new TruckDTO(this.truckID.AsGuid(), this.tare.asFloat(), this.capacity.asFloat(), this.maxBatteryCapacity.asFloat(), this.autonomy.asFloat(), this.fastChargeTime.asFloat());
        }

        public void Update(TruckDTO truckDTO)
        {
            this.truckID = new TruckID(truckDTO.truckID);
            this.tare = new Tare(truckDTO.tare);
            this.capacity = new Capacity(truckDTO.capacity);
            this.maxBatteryCapacity = new MaxBatteryCapacity(truckDTO.maxBatteryCapacity);
            this.autonomy = new Autonomy(truckDTO.autonomy);
            this.fastChargeTime = new FastChargeTime(truckDTO.fastChargeTime);
        }

    }
}

File: Logistics/src/domain/truck/Truck.ts
import { AggregateRoot } from "../../core/domain/AggregateRoot";
import { UniqueEntityID } from "../../core/domain/UniqueEntityID";
import { Result } from "../../core/logic/Result";
import { ITruckDTO } from "../../dto/ITruckDTO";
import { Active } from "./Active";
import { Autonomy } from "./Autonomy";
import { Capacity } from "./Capacity";
import { FastChargeTime } from "./FastChargeTime";
import { MaxBatteryCapacity } from "./MaxBatteryCapacity";
import { Tare } from "./Tare";
import { TruckID } from "./TruckID";


interface TruckProps {
  truckID: TruckID;
  tare: Tare;
  capacity: Capacity;
  maxBatteryCapacity: MaxBatteryCapacity;
  autonomy: Autonomy;
  fastChargeTime: FastChargeTime;
  active: Active;
}

export class Truck extends AggregateRoot<TruckProps> {
  get id (): UniqueEntityID {
    return this._id;
  }

  get truckID (): TruckID {
    return this.props.truckID;
  }

  get capacity (): Capacity {
    return this.props.capacity;
  }

  get maxBatteryCapacity (): MaxBatteryCapacity {
    return this.props.maxBatteryCapacity;
  }

  get autonomy (): Autonomy {
    return this.props.autonomy;
  }

  get fastChargeTime (): FastChargeTime {
    return this.props.fastChargeTime;
  }

  get tare (): Tare {
    return this.props.tare;
  }

  get active (): Active {
    return this.props.active;
  }
  

  set tare (tare: Tare) {
    this.props.tare = tare;
  }

  set capacity (capacity: Capacity) {
    this.props.capacity = capacity;
  }

  set maxBatteryCapacity (maxBatteryCapacity: MaxBatteryCapacity) {
    this.props.maxBatteryCapacity = maxBatteryCapacity;
  }

  set autonomy (autonomy: Autonomy) {
    this.props.autonomy = autonomy;
  }

  set fastChargeTime (fastChargeTime: FastChargeTime) {
    this.props.fastChargeTime = fastChargeTime;
  }

  set active (active: Active) {
    this.props.active = active;
  }
  



  private constructor (props: TruckProps, id?: UniqueEntityID) {
    super(props, id);
  }

  public static create (truckDTO:ITruckDTO , id?: UniqueEntityID): Result<Truck> {

    try {
      const truck = new Truck({
        truckID: TruckID.create(truckDTO.truckID).getValue(),
        tare:Tare.create(truckDTO.tare).getValue(),
        capacity:Capacity.create(truckDTO.capacity).getValue(),
        maxBatteryCapacity:MaxBatteryCapacity.create(truckDTO.maxBatteryCapacity).getValue(),
        autonomy:Autonomy.create(truckDTO.autonomy).getValue(),
        fastChargeTime:FastChargeTime.create(truckDTO.fastChargeTime).getValue(),
        active:Active.create(truckDTO.active).getValue()
      }, id);
  
      return Result.ok<Truck>(truck);

    } catch (error) {
      return Result.fail<Truck>(error);
    }
      
}
      
  



}